packages/blockchain-wallet-v4-frontend/src/components/QRCodeCapture/index.js

```diff
@@ -27,31 +29,51 @@ class QRCodeCaptureContainer extends React.PureComponent {
   }
 
   handleScanBtcAddress (data) {
-    const { address, options } = bip21.decode(data)
-    const { amount, message } = options
-    this.props.formActions.change('sendBtc', 'to', address)
-    this.props.formActions.change('sendBtc', 'amount', amount)
-    this.props.formActions.change('sendBtc', 'message', message)
-    this.props.updateUI({ btcAddress: { toggled: false } })
+    try {
+      const { address, options } = bip21.decode(data)
+      const { amount, message } = options
+      this.props.formActions.change('sendBtc', 'to', address)
+      this.props.formActions.change('sendBtc', 'amount', amount)
+      this.props.formActions.change('sendBtc', 'message', message)
+      this.props.updateUI({ btcAddress: { toggled: false } })
+    } catch (e) {
+      this.props.alertActions.displayError(C.BTC_ADDRESS_INVALID)
```

Comment:
Are we sure that the only thing that could cause issue is an invalid address?

Reply:
Same comment goes for the other scan type methods as well.

Reply:
I kinda am, like the rest is UI change.